app/actions/websocket/index.ts

```diff
@@ -398,3 +409,40 @@ export async function handleEvent(serverUrl: string, msg: WebSocketMessage) {
             break;
     }
 }
+
+async function fetchPostDataIfNeeded(serverUrl: string) {
```

Comment:
moved the code here as:
1. it only affects WS reconnection logic
2. having it in `handleEntryAfterLoadNavigation` was causing database problems with marking the channel as read and the WS updating a pending record (race condition)